src/Geno/C++/Components/Project.cpp

```diff
@@ -68,20 +68,36 @@ void Project::Build( ICompiler& rCompiler )
 	m_FilesLeftToBuild.clear();
 	m_FilesToLink     .clear();
 
-	if( m_Files.empty() )
+	if( m_FileFilters.empty() )
 		return;
 
-	for( const std::filesystem::path& rFile : m_Files )
+	for( const FileFilter& rFileFilter : m_FileFilters )
 	{
-		std::filesystem::path Extension = rFile.extension();
-
-		// #TODO: Compiler will be per-file so this check is only temporary
-		if( Extension == ".cpp"
-		 || Extension == ".cxx"
-		 || Extension == ".cc"
-		 || Extension == ".c" )
+		for( const std::filesystem::path& rFile : rFileFilter.m_Files )
 		{
-			m_FilesLeftToBuild.push_back( rFile );
+			std::filesystem::path Extension = rFile.extension();
+
+			// #TODO: Compiler will be per-file so this check is only temporary
+			if( Extension == ".cpp"
+			 || Extension == ".cxx"
+			 || Extension == ".cc"
+			 || Extension == ".c" )
+			{
+				bool Found = false;
+				for( const std::filesystem::path& rrFile : m_FilesLeftToBuild )
+				{
+					if( std::filesystem::equivalent( rFile, rrFile ) )
+					{
+						Found = true;
+						break;
+					}
+				}
```

Comment:
Does this mean that the same file can be part of multiple filters? Or why do we need to make sure it's not already queued to build?

Reply:
Yeah a single file could be in an *infinite* number of File Filters